script.js

```diff
@@ -49,4 +49,67 @@ const photos = [
     },
 ];
 
-console.log('hello');
\ No newline at end of file
+const photosPerPage = 6;
+let currentIndex = 0;
+
+window.onload = loadInitialPhotos;
+
+function loadInitialPhotos() {
+  currentIndex = 0;
+  renderPhotos(photos);
+}
+
+// render photos
+function renderPhotos(filteredPhotos) {
+  const gallery = document.getElementById('gallery');
+  gallery.innerHTML = '';
+
+  const photosToDisplay = filteredPhotos.slice(0, currentIndex + photosPerPage);
+
+  photosToDisplay.forEach(photo => {
+    const img = document.createElement('img');
+    img.src = photo.url;
+    img.alt = photo.type;
+    gallery.appendChild(img);
+  });
+
+  // hide 'Load More' button if no more photos
+  const loadMoreButton = document.getElementById('loadMore');
+  if (photosToDisplay.length >= filteredPhotos.length) {
+    loadMoreButton.style.display = 'none';
+  } else {
+    loadMoreButton.style.display = 'block';
+  }
+
+  // center "Load More" button
+  const loadMoreContainer = document.querySelector('.load-more');
+  loadMoreContainer.style.display = 'flex';
+  loadMoreContainer.style.justifyContent = 'center';
+}
+
+// load more pictures
+document.getElementById('loadMore').addEventListener('click', () => {
+  currentIndex += photosPerPage;
+  renderPhotos(photos);
+});
+
+// filter by type
+document.getElementById('all').addEventListener('click', () => {
```

Comment:
A suggestion for cleaner code: consider selecting all the buttons as an array, and then using a for loop to set one generic click event function to it. You can determine which filter the button is by using the element's ID in the for loop. That way, you can add new filters without having to change any javascript.